casanovo/config.py

```
@@ -27,185 +27,138 @@
-            elif param_val is not None:
-                self._params[param] = param_type(param_val)
-        except (TypeError, ValueError) as err:
-            logger.error(
-                "Incorrect type for configuration value %s: %s", param, err
-            )
-            raise TypeError(
-                f"Incorrect type for configuration value {param}: {err}"
-            )
+            return _to_python(self._cfg[param])
+        except KeyError:
+            raise AttributeError(f"'Config' object has no attribute '{param}'")
```

Ruff (0.15.6)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)